src/main/java/ch/njol/skript/expressions/ExprBuildRestrictions.java

```diff
@@ -0,0 +1,148 @@
+						Material material = BukkitUnsafe.getMaterialFromMinecraftId(key.toString());
+						if (material != null) {
+							onItem.add(new ItemType(material));
+						}
+					}
+				}
+			}
+		}
+		return onItem.toArray(new ItemType[0]);
+	}
+
+	@SuppressWarnings("NullableProblems")
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
+			return CollectionUtils.array(ItemType[].class);
+		} else if (mode == ChangeMode.RESET || mode == ChangeMode.DELETE) {
+			return CollectionUtils.array();
+		}
+		return null;
+	}
+
+	@SuppressWarnings("ConstantValue")
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		ItemType[] source = getExpr().getArray(event);
+
+		if (mode == ChangeMode.RESET || mode == ChangeMode.DELETE) {
+			for (ItemType item : source) {
+				ItemMeta meta = item.getItemMeta();
+				Collection<Namespaced> empty = new ArrayList<>();
+                if (pattern == DESTROYABLE) { meta.setDestroyableKeys(empty); } else { meta.setPlaceableKeys(empty); }
+                item.setItemMeta(meta);
+			}
+		} else if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
+
+			// only do this if the mode requires it
+			Set<Namespaced> keys = new HashSet<>();
+
+			// get a list of all the provided itemtypes' namespaced keys
+			if (delta != null) {
+				for (Object o : delta) {
+					if (o instanceof ItemType item) {
+						keys.add(item.getRandom().getType().getKey());
+					}
+				}
+			}
+
+			if (mode == ChangeMode.SET) {
+				for (ItemType item : source) {
+					ItemMeta meta = item.getItemMeta();
+                    if (pattern == DESTROYABLE) { meta.setDestroyableKeys(keys); } else { meta.setPlaceableKeys(keys); }
+                    item.setItemMeta(meta);
+				}
+			} else if (mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
+				for (ItemType item : source) {
+					ItemMeta meta = item.getItemMeta();
+					if (pattern == DESTROYABLE ? meta.hasDestroyableKeys() : meta.hasPlaceableKeys()) {
+						Set<Namespaced> alreadyOn = pattern == DESTROYABLE ? meta.getDestroyableKeys() : meta.getPlaceableKeys();
+						if (mode == ChangeMode.ADD) {
+							alreadyOn.addAll(keys);
+						} else if (mode == ChangeMode.REMOVE) {
+							alreadyOn.removeAll(keys);
+						}
+                        if (pattern == DESTROYABLE) { meta.setDestroyableKeys(alreadyOn); } else { meta.setPlaceableKeys(alreadyOn); }
+                    }
+
+					item.setItemMeta(meta);
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean isSingle() {
+		return false;
+	}
+
+	@Override
+	public @NotNull Class<? extends ItemType> getReturnType() {
+		return ItemType.class;
+	}
+
+	@Override
+	public @NotNull String toString(@Nullable Event e, boolean b) {
+		return "build restrictions of " + getExpr().toString(e, b);
```

Comment:
This needs to adapt based on the pattern used (also use `event` and `debug` instead of `e` and `b`.